heartbeat/monitors/active/http/check.go

```diff
@@ -132,14 +135,85 @@ func checkHeaders(headers map[string]string) respValidator {
 	}
 }
 
-func checkBody(matcher []match.Matcher) bodyValidator {
+func parseBody(b interface{}) (positiveMatch, negativeMatch []match.Matcher) {
+	// run through this code block if there is no positive or negative keyword in response body
+	// in this case, there's only plain body
+	if p, ok := b.([]interface{}); ok {
+		for _, pp := range p {
+			if pat, ok := pp.(string); ok {
+				return append(positiveMatch, match.MustCompile(pat)), negativeMatch
+			}
+		}
+	}
+
+	// run through this part if there exists positive/negative keyword in response body
+	// in this case, there will be 3 possibilities: positive + negative / positive / negative
+	if m, ok := b.(map[string]interface{}); ok {
+		for checkType, v := range m {
+			if params, ok := v.([]interface{}); ok {
+				for _, param := range params {
+					if pat, ok := param.(string); ok {
+						if checkType == "positive" {
+							positiveMatch = append(positiveMatch, match.MustCompile(pat))
+						} else if checkType == "negative" {
+							negativeMatch = append(negativeMatch, match.MustCompile(pat))
+						} else {
+							return positiveMatch, negativeMatch
+						}
+					}
+				}
+			}
+		}
+	}
+	return positiveMatch, negativeMatch
+}
+
+/* checkBody accepts 2 parameters:
+1. positive check regex
+2. negative check regex
+*/
+func checkBody(positiveMatch, negativeMatch []match.Matcher) bodyValidator {
+	// in case there's both valid positive and negative regex pattern
 	return func(r *http.Response, body string) error {
-		for _, m := range matcher {
-			if m.MatchString(body) {
+		positiveMatchFlag := false
+		negativeMatchFlag := true
+		// positive match loop
+		for _, pattern := range positiveMatch {
+			if pattern.MatchString(body) {
+				positiveMatchFlag = true
+				break
```

Comment:
Why do we need to bother with the flags when we could just `return` if the positive match fails? That seems a lot simpler.

Reply:
That is in case the customer only give the negative check, for this case, the positive check is always failing.